test/spec/ol/control/control.test.js

```diff
@@ -54,3 +54,25 @@ describe('ol.control.Control\'s target', function() {
     });
   });
 });
+
+describe('ol.control.Control\'s event target', function() {
+  it('is the Control when the Control uses the default target', function() {
+    const ctrl = new Control({element: document.createElement('div')});
+    ctrl.on('test-event', function(e) {
+      expect(e.target).to.be(ctrl);
+    });
+    ctrl.dispatchEvent('test-event');
+    ctrl.dispose();
+  });
+  it('is the Control when the Control has a custom target', function() {
+    const ctrl = new Control({
+      element: document.createElement('div'),
+      target: document.createElement('div')
+    });
+    ctrl.on('test-event', function(e) {
+      expect(e.target).to.be(ctrl);
+    });
+    ctrl.dispatchEvent('test-event');
```

Comment:
Thanks for adding these tests.

As written, this would pass if the event listener were never called.  And if for some reason the listener were called asynchronously, the assertion would be made after the test was complete.  To more reliably make the same assertion, you could declare a `target` before registering the listener, assign to it in the listener, and then make an assertion about the target after dispatching the event.

Reply:
I'd briefly considered that, and decided that it was very unlikely that they'd get called asynchronously. Nevertheless: #10160